tests/test_domain_objects/shared_tests.py

```diff
@@ -0,0 +1,197 @@
+import sys
+from datetime import datetime, timedelta
+
+from test_domain_objects import helper_methods as helper
+from utils.cache import Cache
+from domainobjects import instrument
+
+
+sys.path.insert(0, 'src/')
+
+
+# Shared Tests
+# domain_obj gives access to defined constant lists in the parent class
+# generatable, this is otherwise uninstantiable as it is abstract
+domain_obj = instrument.Instrument(None)
+
+
+def attribute_exists(value, attribute, table):
+    stored = helper.query_db(table, attribute)
+    assert value in stored
+
+
+def swap_contract_id_exists(record):
+    swap_contract_ids = helper.query_db('swap_contracts', 'id')
+    swap_contract_id = record['swap_contract_id']
+    assert swap_contract_id in swap_contract_ids
+
+
+def cusip_exists(record):
+    """ Ensure the generated cusip is of an existing instrument """
+    cusips = helper.query_db('instruments', 'cusip')
+    assert record['cusip'] in cusips
+
+
+def cusip_valid(record):
+    """ CUSIPs must both: be integers, and of length 9 """
+    cusip = record['cusip']
+    assert is_int(cusip) and is_length(9, cusip)
+
+
+def isin_exists(record):
+    isins = helper.query_db('instruments', 'isin')
+    assert record['isin'] in isins
+
+
+def isin_valid(record):
+    """ Valid ISIN is in format '"COI""CUSIP""4"' """
+    coi = record['coi']
+    cusip = str(record['cusip'])
+    expected_isin = coi + cusip + '4'
+    assert record['isin'] == expected_isin
+
+
+def ric_exists(record):
+    rics = helper.query_db('instruments', 'ric')
+    ric = record['ric']
+    assert ric in rics
+
+
+def ric_valid(record):
+    """ RIC Formatting should be a ticker followed by an exchange code.
+    In this use case, exchange codes are currently integers. This test
+    will require updating once this generation approach changes to real-world
+    exchange codes. """
+    ric = record['ric']
+    ticker = record['ticker']
+    id = str(record['instrument_id'])
+    assert valid_ric(ticker, ric, id)
+
+
+def valid_ric(ticker, ric, id):
+    """ Valid RIC is in format '"ticker"."id"' """
+    split_ric = ric.split('.')
+    ticker_ = split_ric[0]
+    id_ = split_ric[1]
+    return ticker == ticker_ and id == id_
+
+
+def unique_ids(records, object_name):
+    id_set = set()
+    id_field_name = object_name+'_id'
+
+    for record in records:
+            id_set.add(record[id_field_name])
+    assert len(records) == len(id_set)
+
+
+def dummy_fields_valid(record, object_name):
+    partial_field_name = object_name+'_field'
+    dummy_field_names = get_dummy_field_names(record, partial_field_name)
+
+    for dummy_field in dummy_field_names:
+        if not is_length(10, record[dummy_field]):
```

Comment:
should length be configurable? (eventually)

Reply:
ISSUE

Reply:
Agreed, issue raised to implement user-configurable dummy fields. Issue #60